plugins/modules/rabbitmq_user.py

```diff
@@ -446,6 +721,91 @@ def has_permissions_modifications(self):
     def has_topic_permissions_modifications(self):
         return self.existing_topic_permissions != self.topic_permissions
 
+    def users_api_url_builder(self, username):
+        return "%s://%s:%s/api/users/%s" % (
+            self.login_protocol,
+            self.login_host,
+            self.login_port,
+            parse.quote(username, ""),
+        )
+
+    def get_permissions_api_url_builder(self, username):
+        return "%s://%s:%s/api/users/%s/permissions" % (
+            self.login_protocol,
+            self.login_host,
+            self.login_port,
+            username,
+        )
+
+    def get_topic_permissions_api_url_builder(self, username):
+        return "%s://%s:%s/api/users/%s/topic-permissions" % (
+            self.login_protocol,
+            self.login_host,
+            self.login_port,
+            username,
+        )
+
+    def permissions_api_url_builder(self, username, vhost):
+        if vhost is None or vhost == "/":
+            vhost = "%2F"
+        return "%s://%s:%s/api/permissions/%s/%s" % (
+            self.login_protocol,
+            self.login_host,
+            self.login_port,
+            vhost,
+            username,
+        )
+
+    def topic_permissions_api_url_builder(self, username, vhost):
```

Comment:
`vhost` should be URL encoded in `topic_permissions_api_url_builder`
because it can be: not only `/` but also `/test` etc.

Reply:
for example I have to run it like this
```
    permissions:
      - vhost: "%2Ftest"
        configure_priv: .*
        read_priv: .*
        write_priv: .*
````
instead of 
```
    permissions:
      - vhost: "/test"
        configure_priv: .*
        read_priv: .*
        write_priv: .*
````